### ISSUE TITLE
Error during dispatching of java.awt.event.InvocationEvent[INVOCATION_DEFAULT,runnable=com.intellij.openapi.editor.impl.EditorFactoryImpl$1$1@2eb7276d,notifier=null,catchExceptions=false,when=1360155657051] on sun.awt.windows.WToolkit@1bfa36ba: Editor of class com.intellij.openapi.editor.impl.EditorImpl hasn't been released:

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Bomb1369/junit-generator-v2#7]
```
Intellij IDEA 12.0.3
Got exception after adding new maven project and reloading project because of 
"language level changed" dialog

Comment from JetBrains: BTW Its a third-party plugin which causes the issue. It 
should dispose editor during releaseResources - 
http://youtrack.jetbrains.com/issue/IDEA-100565


--------------
Error during dispatching of 
java.awt.event.InvocationEvent[INVOCATION_DEFAULT,runnable=com.intellij.openapi.
editor.impl.EditorFactoryImpl$1$1@2eb7276d,notifier=null,catchExceptions=false,w
hen=1360155657051] on sun.awt.windows.WToolkit@1bfa36ba: Editor of class 
com.intellij.openapi.editor.impl.EditorImpl hasn't been released:

com.intellij.openapi.util.TraceableDisposable$DisposalException: Editor of 
class com.intellij.openapi.editor.impl.EditorImpl hasn't been released:
 at com.intellij.openapi.util.TraceableDisposable.throwDisposalError(TraceableDisposable.java:44)
 at com.intellij.openapi.editor.impl.EditorImpl.throwDisposalError(EditorImpl.java:690)
 at com.intellij.openapi.editor.impl.EditorFactoryImpl.throwNotReleasedError(EditorFactoryImpl.java:105)
 at com.intellij.openapi.editor.impl.EditorFactoryImpl.validateEditorsAreReleased(EditorFactoryImpl.java:93)
 at com.intellij.openapi.editor.impl.EditorFactoryImpl$1$1.run(EditorFactoryImpl.java:61)
 at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:251)
 at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:727)
 at java.awt.EventQueue.access$200(EventQueue.java:103)
 at java.awt.EventQueue$3.run(EventQueue.java:688)
 at java.awt.EventQueue$3.run(EventQueue.java:686)
 at java.security.AccessController.doPrivileged(Native Method)
 at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
 at java.awt.EventQueue.dispatchEvent(EventQueue.java:697)
 at com.intellij.ide.IdeEventQueue.d(IdeEventQueue.java:673)
 at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:498)
 at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:333)
 at com.intellij.ide.IdeEventQueue.pumpEventsForHierarchy(IdeEventQueue.java:738)
 at com.intellij.openapi.progress.util.ProgressWindow.startBlocking(ProgressWindow.java:198)
 at com.intellij.openapi.application.impl.ApplicationImpl.runProcessWithProgressSynchronously(ApplicationImpl.java:712)
 at com.intellij.openapi.progress.impl.ProgressManagerImpl.a(ProgressManagerImpl.java:306)
 at com.intellij.openapi.progress.impl.ProgressManagerImpl.runProcessWithProgressSynchronously(ProgressManagerImpl.java:283)
 at com.intellij.openapi.project.impl.ProjectManagerImpl.d(ProjectManagerImpl.java:525)
 at com.intellij.openapi.project.impl.ProjectManagerImpl.convertAndLoadProject(ProjectManagerImpl.java:501)
 at com.intellij.openapi.project.impl.ProjectManagerImpl.loadAndOpenProject(ProjectManagerImpl.java:467)
 at com.intellij.ide.impl.ProjectUtil.openProject(ProjectUtil.java:176)
 at com.intellij.openapi.project.impl.ProjectManagerImpl$15.run(ProjectManagerImpl.java:903)
 at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:333)
 at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:251)
 at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:727)
 at java.awt.EventQueue.access$200(EventQueue.java:103)
 at java.awt.EventQueue$3.run(EventQueue.java:688)
 at java.awt.EventQueue$3.run(EventQueue.java:686)
 at java.security.AccessController.doPrivileged(Native Method)
 at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
 at java.awt.EventQueue.dispatchEvent(EventQueue.java:697)
 at com.intellij.ide.IdeEventQueue.d(IdeEventQueue.java:673)
 at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:498)
 at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:333)
 at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
 at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
 at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:150)
 at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:146)
 at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:138)
 at java.awt.EventDispatchThread.run(EventDispatchThread.java:91)
--------------Creation trace: 
java.lang.Throwable
 at com.intellij.openapi.editor.impl.EditorImpl.<init>(EditorImpl.java:143)
 at com.intellij.openapi.editor.impl.EditorFactoryImpl.a(EditorFactoryImpl.java:182)
 at com.intellij.openapi.editor.impl.EditorFactoryImpl.createEditor(EditorFactoryImpl.java:168)
 at org.intellij.plugins.junitgen.ui.JUnitGeneratorConfiguration.setVmTemplate(JUnitGeneratorConfiguration.java:172)
 at org.intellij.plugins.junitgen.ui.JUnitGeneratorConfiguration.setSettings(JUnitGeneratorConfiguration.java:71)
 at org.intellij.plugins.junitgen.ui.JUnitGeneratorConfiguration.<init>(JUnitGeneratorConfiguration.java:41)
 at org.intellij.plugins.junitgen.ui.JUnitGeneratorConfigurable.createComponent(JUnitGeneratorConfigurable.java:88)
 at com.intellij.openapi.options.newEditor.OptionsEditor$Simple.<init>(OptionsEditor.java:1260)
 at com.intellij.openapi.options.newEditor.OptionsEditor.b(OptionsEditor.java:445)
 at com.intellij.openapi.options.newEditor.OptionsEditor.access$2500(OptionsEditor.java:68)
 at com.intellij.openapi.options.newEditor.OptionsEditor$8$1$1.run(OptionsEditor.java:414)
 at com.intellij.openapi.application.impl.ApplicationImpl.runEdtSafeAction(ApplicationImpl.java:1103)
 at com.intellij.openapi.options.newEditor.OptionsEditor$8$1.run(OptionsEditor.java:408)
 at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:935)
 at com.intellij.openapi.options.newEditor.OptionsEditor$8.run(OptionsEditor.java:406)
 at com.intellij.openapi.application.impl.ApplicationImpl$6.run(ApplicationImpl.java:465)
 at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
 at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
 at java.util.concurrent.FutureTask.run(FutureTask.java:166)
 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
 at java.lang.Thread.run(Thread.java:722)
 at com.intellij.openapi.application.impl.ApplicationImpl$1$1.run(ApplicationImpl.java:153)
--------------Kill trace: 
java.lang.Throwable
 at com.intellij.openapi.util.TraceableDisposable.kill(TraceableDisposable.java:39)
 at com.intellij.openapi.editor.impl.EditorImpl.release(EditorImpl.java:697)
 at com.intellij.openapi.editor.impl.EditorFactoryImpl.releaseEditor(EditorFactoryImpl.java:197)
 at com.intellij.openapi.editor.impl.EditorFactoryImpl.validateEditorsAreReleased(EditorFactoryImpl.java:96)
 at com.intellij.openapi.editor.impl.EditorFactoryImpl$1$1.run(EditorFactoryImpl.java:61)
 at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:251)
 at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:727)
 at java.awt.EventQueue.access$200(EventQueue.java:103)
 at java.awt.EventQueue$3.run(EventQueue.java:688)
 at java.awt.EventQueue$3.run(EventQueue.java:686)
 at java.security.AccessController.doPrivileged(Native Method)
 at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
 at java.awt.EventQueue.dispatchEvent(EventQueue.java:697)
 at com.intellij.ide.IdeEventQueue.d(IdeEventQueue.java:673)
 at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:498)
 at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:333)
 at com.intellij.ide.IdeEventQueue.pumpEventsForHierarchy(IdeEventQueue.java:738)
 at com.intellij.openapi.progress.util.ProgressWindow.startBlocking(ProgressWindow.java:198)
 at com.intellij.openapi.application.impl.ApplicationImpl.runProcessWithProgressSynchronously(ApplicationImpl.java:712)
 at com.intellij.openapi.progress.impl.ProgressManagerImpl.a(ProgressManagerImpl.java:306)
 at com.intellij.openapi.progress.impl.ProgressManagerImpl.runProcessWithProgressSynchronously(ProgressManagerImpl.java:283)
 at com.intellij.openapi.project.impl.ProjectManagerImpl.d(ProjectManagerImpl.java:525)
 at com.intellij.openapi.project.impl.ProjectManagerImpl.convertAndLoadProject(ProjectManagerImpl.java:501)
 at com.intellij.openapi.project.impl.ProjectManagerImpl.loadAndOpenProject(ProjectManagerImpl.java:467)
 at com.intellij.ide.impl.ProjectUtil.openProject(ProjectUtil.java:176)
 at com.intellij.openapi.project.impl.ProjectManagerImpl$15.run(ProjectManagerImpl.java:903)
 at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:333)
 at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:251)
 at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:727)
 at java.awt.EventQueue.access$200(EventQueue.java:103)
 at java.awt.EventQueue$3.run(EventQueue.java:688)
 at java.awt.EventQueue$3.run(EventQueue.java:686)
 at java.security.AccessController.doPrivileged(Native Method)
 at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
 at java.awt.EventQueue.dispatchEvent(EventQueue.java:697)
 at com.intellij.ide.IdeEventQueue.d(IdeEventQueue.java:673)
 at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:498)
 at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:333)
 at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
 at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
 at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:150)
 at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:146)
 at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:138)
 at java.awt.EventDispatchThread.run(EventDispatchThread.java:91)
```

Original issue reported on code.google.com by `sho...@gmail.com` on 7 Feb 2013 at 11:05

**Comment by GoogleCodeExporter** [comment]
```
OK. I will see if I can add this. Must have been a change in version 12.
```

Original comment by `osborn.j...@gmail.com` on 27 Feb 2013 at 2:01
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Found the defect. The editor instances were not getting properly released 
during component closeout.
```

Original comment by `osborn.j...@gmail.com` on 27 Feb 2013 at 8:46
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Released fix in build 2.0.10. please test at your convenience.
```

Original comment by `osborn.j...@gmail.com` on 5 Mar 2013 at 1:41
- Changed state: **Fixed**
